### PR TITLE
Refactor store

### PR DESCRIPTION
**Pull request by nafiz1001**

### CODE REVIEW

File: triton-client/src/components/DatasetCard.tsx [hunk 1]
```diff
@@ -23,13 +24,14 @@ export interface StagingAction {
...
+	const readsetsById = useAppSelector((state) => state.readsetsState.readsetsById)
 	const project = useAppSelector((state) => dataset?.external_project_id ? state.projectsState.projectsById[dataset.external_project_id] : undefined)
 	const constants = useAppSelector(selectConstants)
-	const readsetsByDatasetId = useAppSelector((state) => state.readsetsState.readsetsByDatasetId)
-	const readsetsById = useAppSelector((state) => state.readsetsState.readsetsById)
-  const activeRequest = useMemo<DownloadRequest | undefined>(() => {
-    return dataset?.requests[0] ?? undefined
-  }, [dataset?.requests])
+	const activeRequest = useMemo<DownloadRequest | undefined>(() => {
+		return requests && requests[0] ? requests[0] : undefined
```
Comment (UlysseFG) [inline triton-client/src/components/DatasetCard.tsx:33]: Does request need to be represented by an array ? There is only one request by DatasetId to be expected.

File: triton-client/src/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { createSelector } from "@reduxjs/toolkit"
+import { RequestState } from "./store/requests"
+import { RootState } from "./store/store"
+
+
+
+const selectRequestById = (state: RootState) => state.requestsState.requestById
```
Comment (UlysseFG) [inline triton-client/src/selectors.ts:7]: Can this lead confusion between the request own ID and the datasetID it actually uses ?

File: triton-client/src/store/thunks.ts [hunk 3]
```diff
@@ -31,50 +33,40 @@ export const fetchProjects = () => async (dispatch: AppDispatch, getState: () =>
...
+export const fetchDatasets = (runName: TritonRun['name']) => async (dispatch: AppDispatch, getState: () => RootState) => {
 	try {
-		dispatch(DatasetsStateActions.setLoading(externalProjectId))
-		const datasetIds = Object.values(getState().runsState.runsByName).reduce<number[]>((datasetIDs, run) => {
-			if (run) {
-				datasetIDs.push(...run.datasets)
-			}
-			return datasetIDs
-		}, [])
+		const run = getState().runsState.runsByName[runName]
```
Comment (UlysseFG) [inline triton-client/src/store/thunks.ts:56]: ... ?? [] ?

File: triton-client/src/store/thunks.ts [hunk 4]
```diff
@@ -31,50 +33,40 @@ export const fetchProjects = () => async (dispatch: AppDispatch, getState: () =>
...
+
 		// console.info(datasetIds)
-		const datasets = await apiTriton.listDatasetsByIds(datasetIds)
+		const datasets = await apiTriton.listDatasetsByIds(run.datasets)
 		// console.debug(`Loaded datasets succesfully: ${JSON.stringify(datasets)}`)
-		dispatch(DatasetsStateActions.setDatasetsByProjectId({ projectId: externalProjectId, datasets }))
-		dispatch(ReadsetsStateActions.initializeReadsetsByDatasetIds(datasets))
+		dispatch(DatasetsStateActions.setDatasets(datasets))
+
+		datasets.forEach((dataset) => dispatch(RequestsStateActions.setRequests(dataset.requests)))
```
Comment (UlysseFG) [inline triton-client/src/store/thunks.ts:66]: Mmm this line is pretty suspicious ... why are requests coming from dataset ? If you really want to make request independant move the request listing out of listDatasetsByIds ...

File: triton-client/src/components/DatasetCard.tsx (hunk 1 above)
Comment (nafiz1001) [reply triton-client/src/components/DatasetCard.tsx:33]: Oh right. Forgot about that.

File: triton-client/src/selectors.ts (hunk 2 above)
Comment (nafiz1001) [reply triton-client/src/selectors.ts:7]: Yes.

File: triton-client/src/store/thunks.ts (hunk 4 above)
Comment (nafiz1001) [reply triton-client/src/store/thunks.ts:66]: Yeah. I agree with that too.

Comment (nafiz1001) [reply triton-client/src/store/thunks.ts:66]: Another option is to have a single request attached optionally to dataset.

File: triton-client/src/selectors.ts (hunk 2 above)
Comment (nafiz1001) [reply triton-client/src/selectors.ts:4]: Actually no. I would normally denote explicitly that it is datasetID.

File: triton-server/src/api/api-routes.ts [hunk 5]
```diff
@@ -101,6 +101,23 @@ router.get(
 	})
 )
 
+router.get(
+	'/runs-requests/',
```
Comment (UlysseFG) [inline triton-server/src/api/api-routes.ts:105]: are runs-requests ? not sure about the name ... staging-requests? the relation between runs and requests is unclear at best.

**Review by UlysseFG — APPROVED** [review]

**Pull request merged**